### ISSUE TITLE
Nice Name implementation

### ISSUE THREAD

**Issue by tokejepsen** [pyblish/pyblish-qml#85]
Using the name property of a plugin, display this to the user.

Example of property; https://github.com/pyblish/pyblish-napoleon/blob/master/napoleon/plugins/select_napoleon_instances.py#L19

**Comment by mottosso** [comment]
How about this for an interface?

**Plug-in**

```python
class MyPlugin(..):
   name = "My Plugin"
```

**Instance**

Instances can already be called anything, and will be displayed in the GUI properly.

```python
instance = context.create_instance(name="My Nicest Name")
```

**Context**

To add a custom name for the Context, how about this?

```python
context.set_data("name", "Current file")
```

**Comment by mottosso** [comment]
What do you guys think about this as opposed to `name`?

```python
class MyPlugin(...):
  label = "My Plugin"
```

It's apparently what FTrack does for their "Actions" and thus may be more familiar (?) and to the point?

- https://vimeo.com/124336916

**Comment by tokejepsen** [comment]
> What do you guys think about this as opposed to name?

+1

**Comment by BigRoy** [comment]
> What do you guys think about this as opposed to name?

Short, descriptive and to the point.
+1

**Comment by mottosso** [comment]
Yeah, `None` seems sensible. If label isn't implemented, it'll default to the name of the class as normal.

**Comment by mkolar** [comment]
Label sounds great. Exactly what I'd expect.

**Comment by mottosso** [comment]
Implemented in 0.2.10
